repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -145,4 +150,81 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
   def getSplittableHeaderScala(headerName: String): List[String] = getHeadersScala(headerName).foldLeft(List.empty[String])((list, s) => list ++ s.split(","))
 
   override def getSplittableHeaders(headerName: String): util.List[String] = getSplittableHeaderScala(headerName).asJava
+
+  override def getRequestURL: StringBuffer = requestUrl
+
+  def setRequestURL(url: StringBuffer): Unit = {
+    if (Option(url).isEmpty) throw new IllegalArgumentException("null is not a legal argument to setRequestURL")
+
+    requestUrl = url
+  }
+
+  override def getRequestURI: String = requestUri
+
+  def setRequestURI(uri: String): Unit = {
+    if (Option(uri).isEmpty) throw new IllegalArgumentException("null is not a legal argument to setRequestURI")
+
+    requestUri = uri
+  }
+
+  /**
+    * @return a string representation of the query parameters for this request
+    */
+  override def getQueryString: String = queryString
+
+  /**
+    * @param string the desired query string for this request
+    */
+  def setQueryString(string: String): Unit = {
+    // todo: update parameter map
+    // parse the current query string and remove from parameter map
+    // parse the new query string and add to parameter map
```

Comment:
Is this something we actually want to do? Since the parameter map can be set independently, keeping it in sync with the query string map not be possible.